src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -59,141 +66,150 @@ public Task<RestoreResult> ExecuteAsync()
-            {
-                msbuildOutputFiles = BuildAssetsUtils.GetMSBuildOutputFiles(
-                    _request.Project,
-                    assetsFile,
-                    graphs,
-                    localRepositories,
-                    _request,
-                    assetsFilePath,
-                    _success,
-                    _logger);
-            }
+                if (contextForProject.IsMsBuildBased)
+                {
+                    msbuildOutputFiles = BuildAssetsUtils.GetMSBuildOutputFiles(
+                        _request.Project,
+                        assetsFile,
+                        graphs,
+                        localRepositories,
+                        _request,
+                        assetsFilePath,
+                        _success,
+                        _logger);
+                }
 
-            // If the request is for a lower lock file version, downgrade it appropriately
-            DowngradeLockFileIfNeeded(assetsFile);
+                // If the request is for a lower lock file version, downgrade it appropriately
+                DowngradeLockFileIfNeeded(assetsFile);
 
-            // Revert to the original case if needed
-            await FixCaseForLegacyReaders(graphs, assetsFile, token);
+                // Revert to the original case if needed
+                await FixCaseForLegacyReaders(graphs, assetsFile, token);
 
-            // Write the logs into the assets file
-            var logs = _logger.Errors
-                .Select(l => AssetsLogMessage.Create(l))
-                .ToList();
+                // Write the logs into the assets file
+                var logs = _logger.Errors
+                    .Select(l => AssetsLogMessage.Create(l))
+                    .ToList();
 
-            _success &= !logs.Any(l => l.Level == LogLevel.Error);
+                _success &= !logs.Any(l => l.Level == LogLevel.Error);
 
-            assetsFile.LogMessages = logs;
+                assetsFile.LogMessages = logs;
 
-            if (cacheFile != null)
-            {
-                cacheFile.Success = _success;
-            }
+                if (cacheFile != null)
+                {
+                    cacheFile.Success = _success;
+                }
 
-            restoreTime.Stop();
-            // Create result
-            return new RestoreResult(
-                _success,
-                graphs,
-                checkResults,
-                msbuildOutputFiles,
-                assetsFile,
-                _request.ExistingLockFile,
-                assetsFilePath,
-                cacheFile,
-                cacheFilePath,
-                _request.ProjectStyle,
-                restoreTime.Elapsed);
+                restoreTime.Stop();
+
+                var telemetryEvent = new TelemetryEvent(ProjectRestoreInformation);
+                telemetryEvent[RestoreId.Key] = RestoreId.Value;
+
+                telemetry.TelemetryEvent = telemetryEvent;
```

Comment:
is it intentional that you dont want to raise this event for restore noop case?

Reply:
Good catch